### PR TITLE
Merge Staging release v1.9.0 beta.1 into main

### PR DESCRIPTION
**Pull request by ddimatos**
Merging staging 1.9.0-beta.1 tasks into main branch for the beta release.
Addresses issue #981

### CODE REVIEW

File: docs/source/requirements-single.rst [hunk 1]
```diff
@@ -20,7 +20,7 @@ The controller is where the Ansible engine that runs the playbook is installed.
 Refer to RedHat Ansible Certified Content documentation for more on the `controllers dependencies`_.
 
 .. _controllers dependencies:
-   https://ibm.github.io/z_ansible_collections_doc/requirements/requirements_controller.html
+   https://ibm.github.io/z_ansible_collections_doc/requirements/requirements.html#control-node
 .. ...........................................................................
 .. © Copyright IBM Corporation 2020                                          .
```
Comment (richp405) [inline docs/source/requirements-single.rst:25]: shouldn't this be 2020-2024 on copyright?

File: plugins/module_utils/dd_statement.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) IBM Corporation 2020
+# Copyright (c) IBM Corporation 2020, 2023
```
Comment (richp405) [inline plugins/module_utils/dd_statement.py:1]: maybe 2020-2024?

File: plugins/module_utils/template.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) IBM Corporation 2022 - 2023
+# Copyright (c) IBM Corporation 2022, 2023
```
Comment (richp405) [inline plugins/module_utils/template.py:1]: should be 2024

File: tests/functional/modules/test_zos_tso_command_func.py
Comment (richp405) [inline tests/functional/modules/test_zos_tso_command_func.py:1]: copyright 2024

File: plugins/modules/zos_copy.py [hunk 4]
```diff
@@ -114,7 +114,7 @@
         be deleted and recreated following the process outlined in the C(volume) option.
       - When the C(dest) is an existing VSAM (RRDS), then the source must be an RRDS.
         The VSAM (RRDS) will be deleted and recreated following the process outlined
-        in the C(volume) option.
+       in the C(volume) option.
```
Comment (richp405) [inline plugins/modules/zos_copy.py:117]: This is outdented 1 - the 'i' should be under the 'T'

File: plugins/modules/zos_operator.py [hunk 5]
```diff
@@ -161,7 +164,12 @@
 try:
     from zoautil_py import opercmd
 except Exception:
-    opercmd = MissingZOAUImport()
+    opercmd = ZOAUImportError(traceback.format_exc())
+
+try:
```
Comment (richp405) [inline plugins/modules/zos_operator.py:169]: We added module_utils/zoa_version_checker.py so this would behave more consistently

**Review by richp405 — CHANGES_REQUESTED** [review]
Most comments are about year, but one file needs to use the newer, updated version-checker, so that we are consistent.  It is in zos_operator.py.  I believe it may also be in zos_operator_action_query, which is not in this diff.

File: plugins/module_utils/dd_statement.py (hunk 2 above)
Comment (ddimatos) [reply plugins/module_utils/dd_statement.py:1]: Hey Rich, thanks for looking out for copyrights, most if not all this content was from 2023, template.py's last change was 4 months ago.

BTW, unrelated to your comment, copyrights should be in this form: YEAR1, YEAR2 where YEAR2 is the last year it was updated.

Pulled from our channel:
```
Regarding the copyright year commas 2020, 2021, 2023 vs 2020 - 2023 the suggestion is that it should contain 2 years, the start year where its original creation, and an end year being the last year it was modified, so it should be 2020,2023 
```

File: docs/source/requirements-single.rst (hunk 1 above)
Comment (ddimatos) [reply docs/source/requirements-single.rst:25]: Good point, looks like sphynx only supports one year, so the latest, format they allow is via the config:
```
A copyright statement in the style '2008, Author Name'.

Changed in version 7.1: The value may now be a sequence of copyright statements in the above form, which will be displayed each to their own line.
```

Ill update it to:
```
copyright = '2024, IBM'
```
See doc [link](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-copyright).

File: plugins/module_utils/template.py (hunk 3 above)
Comment (ddimatos) [reply plugins/module_utils/template.py:1]: Hey Rich, thanks for looking out for copyrights, most if not all this content was from 2023, `template.py`'s last change was 4 months ago.

BTW, unrelated to your comment, copyrights should be in this form: YEAR1, YEAR2 where YEAR2 is the last year it was updated.

Pulled from our channel:

Regarding the copyright year commas 2020, 2021, 2023 vs 2020 - 2023 the suggestion is that it should contain 2 years, the start year where its original creation, and an end year being the last year it was modified, so it should be 2020,2023

File: tests/functional/modules/test_zos_tso_command_func.py
Comment (ddimatos) [reply tests/functional/modules/test_zos_tso_command_func.py:1]: Dobe

File: plugins/modules/zos_copy.py (hunk 4 above)
Comment (ddimatos) [reply plugins/modules/zos_copy.py:117]: Good catch, thank you.

File: plugins/modules/zos_operator.py (hunk 5 above)
Comment (ddimatos) [reply plugins/modules/zos_operator.py:169]: Updated 
```
    use_wait_arg = False
    if zoau_version_checker.is_zoau_version_higher_than("1.2.4"):
        use_wait_arg = True
```

**Review by richp405 — APPROVED** [review]
Thank you for all that.

File: CHANGELOG.rst [hunk 6]
```diff
@@ -29,18 +65,18 @@ Minor Changes
...
+- zos_script - Add support for remote_tmp from the Ansible configuration to setup where temporary files will be created, replacing the module option tmp_path. (https://github.com/ansible-collections/ibm_zos_core/pull/1068).
+- zos_tso_command - Add example for executing explicitly a REXX script from a data set. (https://github.com/ansible-collections/ibm_zos_core/pull/1072).
 - zos_unarchive -  Add validation into path joins to detect unauthorized path traversals. (https://github.com/ansible-collections/ibm_zos_core/pull/1029)
 - zos_unarchive - Enhanced test cases to use test lines the same length of the record length. (https://github.com/ansible-collections/ibm_zos_core/pull/965)
 
 Deprecated Features
 -------------------
 
-- zos_blockinfile - debug is deprecated in favor of 'as_json' (https://github.com/ansible-collections/ibm_zos_core/pull/904).
+- zos_blockinfile debug - is deprecated in favor of 'as_json' (https://github.com/ansible-collections/ibm_zos_core/pull/904).
```
Comment (ketankelkar) [inline CHANGELOG.rst:79]: ^ this looks strange -- why was the correctly placed `-` changed here? Seems like

File: changelogs/fragments/1048-update-ac-tool-pyyaml-version.yml
Comment (ketankelkar) [inline changelogs/fragments/1048-update-ac-tool-pyyaml-version.yml:1]: this fragment file name / linked PR is mismatched. The file name is wrong -- should have been `1083-update-ac-tool-pyyaml-version.yml`

File: docs/source/modules/zos_gather_facts.rst
Comment (ketankelkar) [inline docs/source/modules/zos_gather_facts.rst:1]: i'm a bit curious why this one's default value came out to `[]` whereas nearly all of the rest of them have `**default**: False`

File: docs/source/modules/zos_tso_command.rst
Comment (ketankelkar) [inline docs/source/modules/zos_tso_command.rst:1]: i'm a bit curious why this one's default value came out to `0` whereas nearly all of the rest of them have **default**: False

**Review by ketankelkar — APPROVED** [review]
found some curiosities, but nothing objectionable which I can recommend a change on -- therefore this one's approved

File: CHANGELOG.rst (hunk 6 above)
Comment (ddimatos) [reply CHANGELOG.rst:79]: Its a good question, I did not release 1.8 but it looks like it was corrected, so if the changelog fragments still exist  (i mean they always do even when we remove them in GH) but had not been cleaned up, the one who released might have corrected the changelog fragments and generated the changelog again, its a bit of work and have to undo a few things for it to generate over but its possible. 

image

File: changelogs/fragments/1048-update-ac-tool-pyyaml-version.yml
Comment (ddimatos) [reply changelogs/fragments/1048-update-ac-tool-pyyaml-version.yml:1]: Good catch, from a CHANGELOG.RST perspective the content is correct, meaning the text and PR match, the fragment itself will be deleted so it should not cause any issues. To change the fragment name we would need to rename the file, undo changelog meta data, rebuild changelog and check it all back in, so it seems ok to leave it.

File: docs/source/modules/zos_gather_facts.rst
Comment (ddimatos) [reply docs/source/modules/zos_gather_facts.rst:1]: The module seems to have defined the default as an empty list `[]`, so the Jinja template read that and extracted that as RST as it found it.

image

**Pull request merged**